JwtProfile expects audience claim to always be a string

I am using jax-rs-pac4j 1.0.0 with pac4j 1.9.5 and I use Jackson for serializing the response entities. Given the following JAX-RS resource:

```
@Path("/profile")
@Pac4JSecurity(clients = "HeaderClient", authorizers = "isAuthenticated")
public class UserProfileResource {
    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response getUserData(@Pac4JProfile JwtProfile profile) {
        return Response.ok(profile).build();
    }
}
```

When I make a GET request to this endpoint, a ClassCastException is thrown when JwtProfile tries to cast the audience claim into a string. In my case the claim is an array of strings which is valid [according to the RFC spec](https://tools.ietf.org/html/rfc7519#section-4.1.3).

Here is the stack trace:
```
com.fasterxml.jackson.databind.JsonMappingException: java.util.Arrays$ArrayList cannot be cast to java.lang.String (through reference chain: org.pac4j.jwt.profile.JwtProfile["audience"])
	... 
Caused by: java.lang.ClassCastException: java.util.Arrays$ArrayList cannot be cast to java.lang.String
	at org.pac4j.jwt.profile.JwtProfile.getAudience(JwtProfile.java:27)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:664)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:690)
	... 53 common frames omitted
```

--- a/pac4j-core/src/main/java/org/pac4j/core/profile/CommonProfile.java
+++ b/pac4j-core/src/main/java/org/pac4j/core/profile/CommonProfile.java
@@ -1,8 +1,12 @@
 package org.pac4j.core.profile;
 
 import org.pac4j.core.context.Pac4jConstants;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
 
 import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import java.util.Locale;
 
 /**
@@ -108,5 +112,24 @@ public class CommonProfile extends UserProfile {
      */
     public String getLocation() {
         return (String) getAttribute("location");
+    }
+
+    /**
+     * Retrieve the audience (for the JwtProfile and OidcProfile).
+     *
+     * @return the audience
+     */
+    @SuppressWarnings("unchecked")
+    protected List<String> retrieveAudience() {
+        final Object audience = getAttribute(CommonProfileDefinition.AUDIENCE);
+        if (audience instanceof String) {
+            return Collections.singletonList((String) audience);
+        } else if (audience instanceof String[]) {
+            return Arrays.asList((String[]) audience);
+        } else if (audience instanceof List) {
+            return (List<String>) audience;
+        } else {
+            return null;
+        }
     }
 }

--- a/pac4j-core/src/main/java/org/pac4j/core/profile/CommonProfile.java
+++ b/pac4j-core/src/main/java/org/pac4j/core/profile/CommonProfile.java
@@ -25,7 +25,7 @@ public class CommonProfile extends UserProfile {
      * @return the email of the user
      */
     public String getEmail() {
-        return (String) getAttribute("email");
+        return (String) getAttribute(CommonProfileDefinition.EMAIL);
     }
 
     /**
@@ -34,7 +34,7 @@ public class CommonProfile extends UserProfile {
      * @return the first name of the user
      */
     public String getFirstName() {
-        return (String) getAttribute("first_name");
+        return (String) getAttribute(CommonProfileDefinition.FIRST_NAME);
     }
 
     /**
@@ -43,7 +43,7 @@ public class CommonProfile extends UserProfile {
      * @return the family name of the user
      */
     public String getFamilyName() {
-        return (String) getAttribute("family_name");
+        return (String) getAttribute(CommonProfileDefinition.FAMILY_NAME);
     }
 
     /**
@@ -52,7 +52,7 @@ public class CommonProfile extends UserProfile {
      * @return the displayed name of the user
      */
     public String getDisplayName() {
-        return (String) getAttribute("display_name");
+        return (String) getAttribute(CommonProfileDefinition.DISPLAY_NAME);
     }
 
     /**
@@ -70,7 +70,7 @@ public class CommonProfile extends UserProfile {
      * @return the gender of the user
      */
     public Gender getGender() {
-        final Gender gender = (Gender) getAttribute("gender");
+        final Gender gender = (Gender) getAttribute(CommonProfileDefinition.GENDER);
         if (gender == null) {
             return Gender.UNSPECIFIED;
         } else {
@@ -84,7 +84,7 @@ public class CommonProfile extends UserProfile {
      * @return the locale of the user
      */
     public Locale getLocale() {
-        return (Locale) getAttribute("locale");
+        return (Locale) getAttribute(CommonProfileDefinition.LOCALE);
     }
 
     /**
@@ -93,7 +93,7 @@ public class CommonProfile extends UserProfile {
      * @return the url of the picture of the user.
      */
     public URI getPictureUrl() {
-        return (URI) getAttribute("picture_url");
+        return (URI) getAttribute(CommonProfileDefinition.PICTURE_URL);
     }
 
     /**
@@ -102,7 +102,7 @@ public class CommonProfile extends UserProfile {
      * @return the url of the profile of the user.
      */
     public URI getProfileUrl() {
-        return (URI) getAttribute("profile_url");
+        return (URI) getAttribute(CommonProfileDefinition.PROFILE_URL);
     }
 
     /**
@@ -111,7 +111,7 @@ public class CommonProfile extends UserProfile {
      * @return the location of the user
      */
     public String getLocation() {
-        return (String) getAttribute("location");
+        return (String) getAttribute(CommonProfileDefinition.LOCATION);
     }
 
     /**

--- a/pac4j-core/src/main/java/org/pac4j/core/profile/definition/CommonProfileDefinition.java
+++ b/pac4j-core/src/main/java/org/pac4j/core/profile/definition/CommonProfileDefinition.java
@@ -22,6 +22,7 @@ public class CommonProfileDefinition<P extends CommonProfile> extends ProfileDef
     public static final String PICTURE_URL = "picture_url";
     public static final String PROFILE_URL = "profile_url";
     public static final String LOCATION = "location";
+    public static final String AUDIENCE = "aud";
 
     public CommonProfileDefinition() {
         primary(EMAIL, Converters.STRING);

--- a/pac4j-jwt/src/main/java/org/pac4j/jwt/JwtClaims.java
+++ b/pac4j-jwt/src/main/java/org/pac4j/jwt/JwtClaims.java
@@ -12,8 +12,6 @@ public interface JwtClaims {
 
     String SUBJECT         = "sub";
 
-    String AUDIENCE        = "aud";
-
     String EXPIRATION_TIME = "exp";
 
     String NOT_BEFORE      = "nbf";

--- a/pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtProfile.java
+++ b/pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtProfile.java
@@ -4,6 +4,7 @@ import org.pac4j.core.profile.CommonProfile;
 import org.pac4j.jwt.JwtClaims;
 
 import java.util.Date;
+import java.util.List;
 
 /**
  * Represents a user profile based on a JWT.
@@ -23,8 +24,8 @@ public class JwtProfile extends CommonProfile {
         return (String) getAttribute(JwtClaims.ISSUER);
     }
 
-    public String getAudience() {
-        return (String) getAttribute(JwtClaims.AUDIENCE);
+    public List<String> getAudience() {
+        return retrieveAudience();
     }
 
     public Date getExpirationDate() {

--- a/pac4j-oidc/src/main/java/org/pac4j/oidc/profile/OidcProfile.java
+++ b/pac4j-oidc/src/main/java/org/pac4j/oidc/profile/OidcProfile.java
@@ -83,14 +83,8 @@ public class OidcProfile extends CommonProfile {
         return (String) getAttribute(OidcProfileDefinition.ISSUER);
     }
 
-    @SuppressWarnings("unchecked")
     public List<String> getAudience() {
-        final Object audience = getAttribute(OidcProfileDefinition.AUDIENCE);
-        if (audience instanceof String) {
-            return Collections.singletonList((String) audience);
-        } else {
-            return (List<String>) audience;
-        }
+        return retrieveAudience();
     }
 
     public Date getExpirationDate() {

--- a/pac4j-oidc/src/main/java/org/pac4j/oidc/profile/OidcProfileDefinition.java
+++ b/pac4j-oidc/src/main/java/org/pac4j/oidc/profile/OidcProfileDefinition.java
@@ -39,7 +39,6 @@ public class OidcProfileDefinition<P extends OidcProfile> extends CommonProfileD
     public static final String REFRESH_TOKEN = "refresh_token";
     public static final String ISSUER          = "iss";
     public static final String SUBJECT         = "sub";
-    public static final String AUDIENCE        = "aud";
     public static final String EXPIRATION_TIME = "exp";
     public static final String ISSUED_AT       = "iat";
     public static final String AUTH_TIME       = "auth_time";

